### PR TITLE
Feat archive project

### PR DESCRIPTION
**Pull request by JakubKostrzewsky**
Archive project.

## Description

From now User can archive projects.

## Type of Change

- [x] ✨ New feature (non-breaking change which adds functionality)
- [ ] 🛠️ Bug fix (non-breaking change which fixes an issue)
- [ ] ❌ Breaking change (fix or feature that would cause existing functionality to change)
- [ ] 🧹 Code refactor
- [ ] ✅ Build configuration change
- [ ] 📝 Documentation
- [ ] 🗑️ Chore

### CODE REVIEW

File: lib/features/archive_project/dependencies.dart [hunk 1]
```diff
@@ -0,0 +1,15 @@
+import 'package:appunite_internal_app/features/archive_project/data/repositories/archive_project_repository_impl.dart';
+import 'package:appunite_internal_app/features/archive_project/domain/repositories/archive_project_repository.dart';
+import 'package:appunite_internal_app/features/archive_project/domain/use_cases/archive_project_use_case.dart';
+import 'package:appunite_internal_app/features/archive_project/presentation/bloc/archive_project_cubit.dart';
+
+import '../../dependencies.dart';
+
+void setup() {
+  // Data
+  sl.registerFactory<ArchiveProjectRepository>(() => ArchiveProjectRepositoryImpl(sl()));
```
Comment (JakubKostrzewsky) [inline lib/features/archive_project/dependencies.dart:10]: Make this a singleton

File: test/features/archive_project/presentation/bloc/archive_project_cubit_test.dart [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+
+    blocTest<ArchiveProjectCubit, ArchiveProjectState>(
+      'emits [loading, failed] when user archive project unsuccessful',
+      build: () {
+        when(() => archiveProjectUseCase.execute(any())).thenAnswer((_) async => const Left(failure));
+
+        return ArchiveProjectCubit(archiveProjectUseCase);
+      },
+      act: (ArchiveProjectCubit bloc) => bloc.archiveProject(projectId),
+      expect: () => <ArchiveProjectState>[
```
Comment (JakubKostrzewsky) [inline test/features/archive_project/presentation/bloc/archive_project_cubit_test.dart:54]: Make whole list as const

File: lib/core/dependencies.dart [hunk 3]
```diff
@@ -1,10 +1,39 @@
...
+          }
+        },
+      ),
+    );
+
+  (dio.httpClientAdapter as DefaultHttpClientAdapter).onHttpClientCreate = (HttpClient client) {
+    client.badCertificateCallback = (X509Certificate cert, String host, int port) => true;
+
+    return client;
+  };
```
Comment (mhnowak) [inline lib/core/dependencies.dart:34]: It looks like it's only temporary, is it?

File: lib/core/http_client.dart [hunk 4]
```diff
@@ -1,17 +1,14 @@
 import 'package:appunite_internal_app/core/error_handler.dart';
 import 'package:dio/dio.dart';
-import 'package:flutter/material.dart';
 
-class HttpClient {
-  final Dio _client = Dio();
+class AppUniteHttpClient {
+  final Dio client;
```
Comment (mhnowak) [inline lib/core/http_client.dart:5]: why public now?

File: lib/core/presentation/widgets/failure.dart [hunk 5]
```diff
@@ -0,0 +1,17 @@
+import 'package:appunite_internal_app/core/domain/entities/failure.dart';
+import 'package:flutter/material.dart';
+
+class FailurePage extends StatelessWidget {
+  const FailurePage(this.error);
```
Comment (mhnowak) [inline lib/core/presentation/widgets/failure.dart:5]: Think about making it a named parameter. All material widgets use this convention so I think we should as well.

**Pull request merged**